packages/calling/src/CallingClient/line/line.test.ts

```diff
@@ -0,0 +1,203 @@
+import {Mutex} from 'async-mutex';
+import {
+  getMobiusDiscoveryResponse,
+  getMockDeviceInfo,
+  getMockRequestTemplate,
+  getTestUtilsWebex,
+} from '../../common/testUtil';
+import {URL} from '../registration/registerFixtures';
+import {filterMobiusUris} from '../../common';
+import {
+  MobiusServers,
+  MobiusStatus,
+  ServiceIndicator,
+  WebexRequestPayload,
+} from '../../common/types';
+import {LineStatus} from './types';
+import Line from '.';
+import * as utils from '../../common/Utils';
+import SDKConnector from '../../SDKConnector';
+import {URL_ENDPOINT} from '../constants';
+import {LOGGER} from '../../Logger/types';
+import * as regUtils from '../registration/register';
+
+describe('Line Tests', () => {
+  const mutex = new Mutex();
+  const webex = getTestUtilsWebex();
+  SDKConnector.setWebex(webex);
+
+  const defaultServiceData = {indicator: ServiceIndicator.CALLING, domain: ''};
+  const createRegistrationSpy = jest.spyOn(regUtils, 'createRegistration');
+
+  const mobiusUris = filterMobiusUris(getMobiusDiscoveryResponse(), URL);
+  const primaryMobiusUris = jest.fn(() => mobiusUris.primary);
+  const backupMobiusUris = jest.fn(() => mobiusUris.backup);
+  const userId = webex.internal.device.userId;
+  const clientDeviceUri = webex.internal.device.url;
+
+  const handleErrorSpy = jest.spyOn(utils, 'handleRegistrationErrors');
+
+  jest.clearAllMocks();
+
+  describe('Registration tests', () => {
```

Comment:
Looks like `case LINE_EVENTS.ERROR:` lineEmitter method is never hit. Can we write a testcase for that?

Reply:
registration failure cases should be hitting the LINE_EVENTS.ERROR only

Reply:
Taken care of this line. Please check. Updated the coverage report screenshot as well.